catch-uncommitted.sh

```diff
@@ -39,7 +39,7 @@ if [ $catch_no_git -eq 1 ] && !(hash git 2> /dev/null); then
 elif ! git diff HEAD --exit-code ${ignore_eol} -- . "${extra_args[@]}" ; then
 	echo '** Uncommitted changes found (^^^ diff above ^^^) - FAIL **'
 	exit 1
-elif test -n "$(git ls-files --exclude-standard --others | tee /dev/tty)" ; then
+elif test -n "$(git ls-files --exclude-standard --others)" ; then
```

Comment:
I'm not sure *why* the `/dev/tty` stuff is there, but presumably for a reason. If you don't want to tee to `/dev/tty` when it doesn't exist, then this change would be backwards compatible with the current behaviour:

```zsh
gitpipe=(tee /dev/tty)
[ ! -w /dev/tty ] && gitpipe=(cat)
# ...
elif test -n "$(git ls-files --exclude-standard --others | "${gitpipe[@]} )" ; then
```

Reply:
@wrboyce Not sure why I didn't think of this! I'll add this in and see how it goes. :+1: